### PR TITLE
 LTD-2816-db-changes-security-approval

### PR DESCRIPTION
**Pull request by depsiatwal**

### CODE REVIEW

File: api/applications/models.py [hunk 1]
```diff
@@ -226,7 +227,17 @@ class StandardApplication(BaseApplication):
     goods_recipients = models.TextField(choices=GOODS_RECIPIENTS_CHOICES, default="")
     goods_starting_point = models.TextField(choices=GOODS_STARTING_POINT_CHOICES, default="")
 
+    # MOD Security approval fields
     is_mod_security_approved = models.BooleanField(blank=True, default=None, null=True)
```
Comment (currycoder) [inline api/applications/models.py:231]: Just triple checking, but this is where the answer for the first security approval question will be stored, right?

Comment (depsiatwal) [reply api/applications/models.py:231]: that's correct :-) it's the one you created

Comment (kevincarrogan) [reply api/applications/models.py:231]: Isn't this actually a little redundant though? Can't the value of this be inferred from `security_approvals`?

File: api/applications/models.py [hunk 2]
```diff
@@ -226,7 +227,17 @@ class StandardApplication(BaseApplication):
...
+    security_approvals = ArrayField(
+        models.CharField(choices=SecurityClassifiedApprovalsType.choices, max_length=255), blank=True, null=True
+    )
+    f680_reference_number = models.TextField(default=None, blank=True, null=True, max_length=100)
+    f1686_contracting_authority = models.TextField(default=None, blank=True, null=True, max_length=100)
+    is_f1686_approval_document_available = models.BooleanField(blank=True, default=None, null=True)
+    f1686_reference_number = models.TextField(default=None, blank=True, null=True, max_length=100)
+    f1686_approval_date = models.DateField(blank=False, null=True)
+    other_security_approval_details = models.TextField(default=None, blank=True, null=True, max_length=2500)
```
Comment (kevincarrogan) [inline api/applications/models.py:240]: I think we've tried to move away from having `max_length` for `TextField` entries.

File: api/applications/models.py [hunk 3]
```diff
@@ -226,7 +227,17 @@ class StandardApplication(BaseApplication):
...
 
+    # MOD Security approval fields
     is_mod_security_approved = models.BooleanField(blank=True, default=None, null=True)
+    security_approvals = ArrayField(
+        models.CharField(choices=SecurityClassifiedApprovalsType.choices, max_length=255), blank=True, null=True
+    )
+    f680_reference_number = models.TextField(default=None, blank=True, null=True, max_length=100)
+    f1686_contracting_authority = models.TextField(default=None, blank=True, null=True, max_length=100)
+    is_f1686_approval_document_available = models.BooleanField(blank=True, default=None, null=True)
+    f1686_reference_number = models.TextField(default=None, blank=True, null=True, max_length=100)
```
Comment (kevincarrogan) [inline api/applications/models.py:238]: I think this makes more sense to be a `CharField`.

File: api/applications/models.py [hunk 4]
```diff
@@ -226,7 +227,17 @@ class StandardApplication(BaseApplication):
     goods_recipients = models.TextField(choices=GOODS_RECIPIENTS_CHOICES, default="")
     goods_starting_point = models.TextField(choices=GOODS_STARTING_POINT_CHOICES, default="")
 
+    # MOD Security approval fields
     is_mod_security_approved = models.BooleanField(blank=True, default=None, null=True)
+    security_approvals = ArrayField(
+        models.CharField(choices=SecurityClassifiedApprovalsType.choices, max_length=255), blank=True, null=True
+    )
+    f680_reference_number = models.TextField(default=None, blank=True, null=True, max_length=100)
```
Comment (kevincarrogan) [inline api/applications/models.py:235]: I think this makes more sense to be a `CharField`.

File: api/applications/enums.py [hunk 5]
```diff
@@ -196,3 +196,30 @@ def get_text(cls, choice):
...
+
+    choices = (
+        (F680, "F680"),
+        (F1686, "F1686"),
+        (OTHER, "Other"),
+    )
+
+    @classmethod
+    def keys(cls):
+        return [key for key, _ in cls.choices]
```
Comment (kevincarrogan) [inline api/applications/enums.py:225]: For these new enums I think we should just start using `models.TextChoices` https://docs.djangoproject.com/en/4.1/ref/models/fields/#enumeration-types

File: api/applications/models.py (hunk 1 above)
Comment (depsiatwal) [reply api/applications/models.py:231]: I did think about this and for the other we refers this

security_approvals is an array I just think it's cleaner and easier for UI to set this and BE logic to use this field.

File: api/applications/models.py (hunk 2 above)
Comment (depsiatwal) [reply api/applications/models.py]: done

File: api/applications/models.py (hunk 3 above)
Comment (depsiatwal) [reply api/applications/models.py]: ahh yes changed except details field

File: api/applications/models.py (hunk 4 above)
Comment (depsiatwal) [reply api/applications/models.py]: done

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**